### PR TITLE
feat(notifier): adds provider validation

### PR DESCRIPTION
**Pull request by vjgene**
Adds provider url validation during registration

### CODE REVIEW

File: src/models/UIMessage.ts [hunk 1]
```diff
@@ -11,7 +12,8 @@ export interface CustomAction {
     action: Function
 }
 
-export type ErrorId = APIErrorId | ContractErrorId | MarketErrorId
+export type ErrorId = APIErrorId | ContractErrorId |
+    MarketErrorId | NotifierErrorId
```
Comment (jurajpiar) [inline src/models/UIMessage.ts]: Remove from here. It's too domain specific for this place.

File: src/components/organisms/notifier/ProviderRegistrar.tsx [hunk 2]
```diff
@@ -23,6 +26,21 @@ const ProviderRegistrar: FC<ProviderRegistrarProps> = ({
...
   const { register, handleSubmit, errors } = useForm<Inputs>()
+  const reportError = useErrorReporter()
+
+  const validateProviderURL = async (url: string): Promise<boolean> => {
+    const notifierService: ProviderValidationService = new ProviderValidationService(url)
+    notifierService.connect(reportError)
+    const result: string = await notifierService.healthCheck().catch((error) => {
+      reportError({
+        error,
+        id: 'notifier-provider',
```
Comment (jurajpiar) [inline src/components/organisms/notifier/ProviderRegistrar.tsx]: that's just an api error type. the text should be used to narrow down the location of the error.

File: src/api/rif-notifier-service/provider.ts [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+      if (content.length) return message
+      throw new UIError({
+        error: new Error('No available subscription plan'),
+        text: 'Atleast one subscription plan must be available for registration',
+        id: 'service-fetch',
+      })
+    }
+    return API_RESPONSE_MESSAGES.ERROR
+  }
```
Comment (jurajpiar) [inline src/api/rif-notifier-service/provider.ts]: how come you didn't just add and endpoint to the existing api file?

**Review by jurajpiar — DISMISSED** [review]
Tiny things though I feel I you might have misunderstood me when I said to just use the existing service for your endpoint.

File: src/api/rif-notifier-service/provider.ts (hunk 3 above)
Comment (vjgene) [reply src/api/rif-notifier-service/provider.ts]: This i wanted to separate validation logic from core notification service

**Review by jurajpiar — APPROVED** [review]
LGTM

File: src/components/organisms/notifier/ProviderRegistrar.tsx [hunk 4]
```diff
@@ -52,7 +70,10 @@ const ProviderRegistrar: FC<ProviderRegistrarProps> = ({
                   inputProps={{
                     style: { textAlign: 'center' },
                   }}
-                  inputRef={register({ required: true })}
+                  inputRef={register({
+                    required: true,
+                    validate: { validateProviderURL },
```
Comment (itofarina) [inline src/components/organisms/notifier/ProviderRegistrar.tsx:67]: 😍

File: src/constants/strings.ts [hunk 5]
```diff
@@ -10,4 +10,9 @@ and still undergoing final testing before the official release.
   and any purchasing of storage space should be done
    at the user’s own risk.`
 
+export const API_RESPONSE_MESSAGES = {
+  OK: 'OK',
+  ERROR: 'error',
+}
```
Comment (itofarina) [inline src/constants/strings.ts:16]: Could it be better to have it as
`type ApiResponse = 'OK' | 'ERROR'`?

File: src/components/organisms/notifier/ProviderRegistrar.tsx [hunk 6]
```diff
@@ -23,6 +26,21 @@ const ProviderRegistrar: FC<ProviderRegistrarProps> = ({
   providerAddress, onRegister, isEnabled,
 }) => {
   const { register, handleSubmit, errors } = useForm<Inputs>()
+  const reportError = useErrorReporter()
+
+  const validateProviderURL = async (url: string): Promise<boolean> => {
+    const notifierService: ProviderValidationService = new ProviderValidationService(url)
+    notifierService.connect(reportError)
+    const result: string = await notifierService.healthCheck().catch((error) => {
```
Comment (itofarina) [inline src/components/organisms/notifier/ProviderRegistrar.tsx]: I would prefer to not use string return and use a type instead like:
`result: ApiResponse = ...`

**Review by itofarina — APPROVED** [review]
Apart from the minor `type` comments looks great!

File: src/constants/strings.ts (hunk 5 above)
Comment (jurajpiar) [reply src/constants/strings.ts:16]: either that or better still,
```suggestion
export const API_RESPONSE_STATUS = {
  OK: 'OK',
  ERROR: 'error',
}

export type ApiResponse = keyof typeof API_RESPONSE_STATUS
```

File: src/components/organisms/notifier/ProviderRegistrar.tsx (hunk 6 above)
Comment (jurajpiar) [reply src/components/organisms/notifier/ProviderRegistrar.tsx]: if you are going to do apply this suggestion Vijay, then you might as well do it properly and move the logic part to a mapping.
Also I've just realised that the service is misused a bit. Let's have a call about it.

**Review by jurajpiar — CHANGES_REQUESTED** [review]

**Review by jurajpiar — APPROVED** [review]

**Pull request merged**